### DISCUSSION TITLE
Improve the UI on the orders/new payment screen

### DISCUSSION THREAD

**Discussion by Mandily** [solidusio/solidus#4562] [New Features or Ideas]
I'd like to suggest a few upgrades and layout changes to the "New payment" screen to improve the usability of the page. (The red dotted lines in the wireframe are boundaries to the main content area and sidebar)

1. Display "Update" button in header and at bottom of page for easy access
 
2. Change radio buttons into a dropdown. This will help keep the user focused on which option they've chosen as it directly affects what's populated below.
 
3. Add headings to help user scan form
 
4. Change label from "Name" to "Name on card" for better specificity of what's required
 
5. Change "Card code" to "Security code" for a more generic label (sometimes this is CSV or CVV)
 
6. Add a dropdown to select existing addresses or a new address. Only show this if the user has at least one saved address. If a saved address is chosen, that data gets populated in the respective fields. (This will allow a user to tweak a saved address to create a new one if needed)
 
7. Change "State" to "Region" to be more generic. We could add helpful information like "State, province, country etc" into a "i" tooltip if needed.

![payments no notes pdf](https://cloud.githubusercontent.com/assets/15271353/26311815/80864e58-3efd-11e7-9d85-f86f9d24ebfb.png)

**Comment by mamhoff** [comment]
Hm, I'm not sure we need the billing address on the payment screen. We already have it on the Customer Details screen, don't we? 

Also: I'm missing the ability to use an existing payment source (credit card). Maybe it makes sense to make that one of those [tiered select boxes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup)?

**Comment by Mandily** [comment]
My bad, I was looking at some functionality on an address form on another site and got muddled when I was making this wire. This is more what I had in mind:

![payment option1](https://cloud.githubusercontent.com/assets/15271353/26359636/e2b8a15e-3fcd-11e7-98a7-d856e34fe944.png)

Thinking on it again today, I did come up with another option.
Alternatively we could also put that form into a modal. I realize we don't use that model of interaction right now, but it's something going forward I'd like to do more of instead of routing the user to a new page.
![payment option 2](https://cloud.githubusercontent.com/assets/15271353/26359704/1677bb24-3fce-11e7-95a7-91dfbb6fe4fc.png)

@mtomov What do you think of these options? It's hard for me to say which one is "better" without user input. Do you have any insight from working with clients?
